src/test/codegen/repr-transparent.rs

```diff
@@ -108,7 +108,7 @@ struct f32x4(f32, f32, f32, f32);
 #[repr(transparent)]
 pub struct Vector(f32x4);
 
-// CHECK: define <4 x float> @test_Vector(<4 x float> %arg0)
+// CHECK: define <4 x float> @test_Vector(<4 x float>* %arg0)
 #[no_mangle]
 pub extern fn test_Vector(_: Vector) -> Vector { loop {} }
```

Comment:
Is this change *really* correct?

Reply:
It seems wrong to affect the C ABI, or any non-Rust ABI. We deliberately only pass vectors in memory for the Rust ABI, so in a sense this pass is currently doing far more than undoing argument promotion.

Reply:
I [mentioned below](https://github.com/rust-lang/rust/pull/55073#issuecomment-430267183), but passing SIMD values in FFI is unstable and not supported in rustc right now. It's correct that this demotion is blindly demoting all arguments, irrespective of ABI. That's not affecting stable users, however. I do not personally know of an easy fix for this (but would be willing to implement one!), and would prefer to not block the patch on a bug in what seems to be a rarely-used unstable feature